executor/src/witgen/sequence_iterator.rs

```diff
@@ -201,15 +206,23 @@ impl Iterator for ProcessingSequenceIterator {
         match self {
             Self::Default(it) => it.next(),
             Self::Cached(it) => it.next(),
+            Self::Incomplete => unreachable!(),
```

Comment:
why `unreachable!` instead of `None`?

Reply:
The "Incomplete" sequence iterator means we already know that we won't be able to complete successfully, so the caller should catch that and not even try. Otherwise the sequence iterator is probably used correctly.